### PR TITLE
Adding end-to-end integration test for pipeline

### PR DESCRIPTION
**Pull request by AustinTag**
*Issue #, if available:*

*Description of changes:*
- Adding integration test source set, dependencies, and task
- Adding ES gradle plugin so that we can start and stop a local ES instance for the integration tests
- Adding integration test that sends data through a simple pipeline with 
   - Source: Otel trace source
   - Processors: [OtelRawProcessor]
   - Sinks [ESSink]
- Verifying the correct data in ES at the end using RestHighLevelClient

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kowshikn** [comment]
Can we not build and use an image of situp instead? Something like 
* run an ODFE docker
* run situp dockert with trace analytics pipeline 
* then run the e2e

File: situp-core/build.gradle [hunk 1]
```diff
@@ -1,5 +1,23 @@
 plugins {
     id 'jacoco'
+    id "cgoit.gradle.elasticsearch" version "0.2.3"
```
Comment (chenqi0805) [inline situp-core/build.gradle:3]: Is there a reason using fork instead of the source?

File: situp-core/build.gradle [hunk 2]
```diff
@@ -31,4 +49,43 @@ jar {
...
+
+task integrationTest(type: Test) {
+    description = 'Runs the integration tests.'
+    group = 'verification'
+    testClassesDirs = sourceSets.integrationTest.output.classesDirs
+    classpath = sourceSets.integrationTest.runtimeClasspath
+
+
+    doFirst {
+        startElasticsearch {
```
Comment (chenqi0805) [inline situp-core/build.gradle:62]: QUES: just curious, I assume this local cluster is elasticsearch-oss? Is it possible for this framework to configure the ES to install plugins? For sink, ISM and security plugins are relevant although maybe not in the scope of this e2e test.

File: situp-core/integrationTest.gradle [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * End to end test. Spins up ODFE and situp docker containers, then runs the integ test
+ * Stops the docker containers when finished
+ */
+task endToEndTest(type: Test) {
+    dependsOn build
+    dependsOn startOdfeDockerContainer
+    dependsOn startSitupDockerContainer
+    startSitupDockerContainer.mustRunAfter 'startOdfeDockerContainer'
+
+    description = 'Runs the integration tests.'
```
Comment (kowshikn) [inline situp-core/integrationTest.gradle:107]: nit: call it situp or situp-core integration test, will make debug easier.

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                SearchSourceBuilder.searchSource().size(100)
+        );
+        final SearchResponse searchResponse = restHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
+        final List<Map<String, Object>> foundSources = getSourcesFromSearchHits(searchResponse.getHits());
+        Assert.assertEquals(expectedDocuments.size(), foundSources.size());
+        expectedDocuments.forEach(expectedDoc -> Assert.assertTrue(foundSources.contains(expectedDoc)));
+    }
+
+    private void sendExportTraceServiceRequestToSource(ExportTraceServiceRequest request) throws InvalidProtocolBufferException {
+        WebClient.of().execute(RequestHeaders.builder()
```
Comment (kowshikn) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:90]: I would prefer using gRPC client?

Note: In the longer we should support both gRPC and HTTP/JSON.

File: situp-core/src/integrationTest/resources/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+raw-pipeline:
+  source:
+    otel_trace_source:
+  processor:
```
Comment (kowshikn) [inline situp-core/src/integrationTest/resources/pipeline.yml:4]: We should have service-map also part of CI. Maye be next PR.

File: situp-core/build.gradle [hunk 6]
```diff
@@ -31,4 +35,8 @@ jar {
         configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
     }
 
-}
\ No newline at end of file
+    exclude 'META-INF/*.RSA'
+    exclude 'META-INF/*.SF'
+    exclude 'META-INF/*.DSA'
```
Comment (kowshikn) [inline situp-core/build.gradle:40]: I merged my PR, this should be a merge conflict.

File: situp-core/integrationTest.gradle (hunk 3 above)
Comment (AustinTag) [reply situp-core/integrationTest.gradle:107]: Sure

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java (hunk 4 above)
Comment (AustinTag) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:90]: Sure, I can swap to grpc

File: situp-core/src/integrationTest/resources/pipeline.yml (hunk 5 above)
Comment (AustinTag) [reply situp-core/src/integrationTest/resources/pipeline.yml:4]: Yeah, I will add service map in the next PR

File: situp-core/build.gradle (hunk 6 above)
Comment (AustinTag) [reply situp-core/build.gradle:40]: Sure, i'll look out for that

**Review by kowshikn — APPROVED** [review]

**Review by chenqi0805 — APPROVED** [review]
LGTM!

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+        final ExportTraceServiceRequest exportTraceServiceRequest2 = getExportTraceServiceRequest(
+                getRandomResourceSpans(10)
+        );
+
+        sendExportTraceServiceRequestToSource(exportTraceServiceRequest1);
+        sendExportTraceServiceRequestToSource(exportTraceServiceRequest2);
+
+        //Wait for data to flow through pipeline and be indexed by ES
+        Thread.sleep(2000);
```
Comment (chenqi0805) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:67]: I think now we have better strategy for this:

https://github.com/opendistro-for-elasticsearch/simple-ingest-transformation-utility-pipeline/blob/8c592850813901b0abb5666f7249d8b1720a14c6/situp-plugins/elasticsearch/src/test/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSinkIT.java#L242

checking if it works

Comment (chenqi0805) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:67]: Made a patching PR with the relevant change to merge into this one.

**Pull request merged**